app/src/main/java/com/github/se/icebreakrr/ui/map/LocationSelectorMap.kt

```diff
@@ -0,0 +1,201 @@
+  val centerLongitude =
+      profilesViewModel.selfProfile.value?.location?.longitude ?: DEFAULT_USER_LONGITUDE
+  var markerState by remember { mutableStateOf<MarkerState?>(null) }
+  val profileId = if (!isTesting) navBackStackEntry?.arguments?.getString("userId") else "2"
+  val profile = profilesViewModel.selectedProfile.collectAsState()
+  val context = LocalContext.current
+  val lastKnownLocation = locationViewModel.lastKnownLocation.collectAsState()
+  var mapLoaded by remember { mutableStateOf(false) }
+  var stringQuery by remember { mutableStateOf("") }
+
+  LaunchedEffect(Unit) {
+    profilesViewModel.getSelfProfile() {}
+    profilesViewModel.getProfileByUid(profileId ?: "null")
+    markerState =
+        MarkerState(
+            position =
+                LatLng(
+                    lastKnownLocation.value?.latitude ?: DEFAULT_USER_LATITUDE,
+                    lastKnownLocation.value?.longitude ?: DEFAULT_USER_LONGITUDE))
+  }
+
+  val cameraPositionState = rememberCameraPositionState {
+    position = CameraPosition.fromLatLngZoom(LatLng(centerLatitude, centerLongitude), DEFAULT_ZOOM)
+  }
+
+  Scaffold(
+      modifier =
+          Modifier.fillMaxSize().background(Color.White).testTag("LocationSelectorMapScreen"),
+      topBar = { TopBar("Select Meeting Point", false, {}) },
+      bottomBar = {
+        Box(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .height(BOTTOM_BAR_HEIGHT.dp)
+                    .background(color = IceBreakrrBlue)
+                    .testTag("addTextAndSendLocationBox"),
+        ) {
+          TextField(
+              value = stringQuery,
+              onValueChange = {
+                if (it.length < TEXTFIELD_MAX_CHAR) {
+                  stringQuery = it
+                }
+              },
+              label = { Text("Add Details...", modifier = Modifier.testTag("labelTagSelector")) },
+              modifier =
+                  Modifier.padding(
+                          start = TEXT_FIELD_HORIZONTAL_PADDING.dp,
+                          end = TEXT_FIELD_HORIZONTAL_PADDING.dp,
+                          top = TEXT_FIELD_VERTICAL_PADDING.dp,
+                          bottom = TEXT_FIELD_VERTICAL_PADDING.dp)
+                      .fillMaxSize()
+                      .shadow(
+                          elevation = TEXT_FIELD_ELEVATION.dp,
+                          shape = RoundedCornerShape(TEXT_FIELD_ROUNDED_CORNER.dp))
+                      .testTag("addDetailsTextField"),
+          )
+          IconButton(
+              onClick = {
+                if (mapLoaded || isTesting) {
+                  meetingRequestViewModel.confirmMeetingLocation(
+                      profileId!!,
+                      Pair(
+                          stringQuery,
+                          Pair(
+                              markerState?.position?.latitude!!,
+                              markerState?.position?.longitude!!)))
```

Comment:
"!!" could lead to crashes if the values are null, try adding more none null logic to the code

Reply:
Okay I see now thanks